libsolutil/Common.h

```diff
@@ -73,6 +73,9 @@ using StringMap = std::map<std::string, std::string>;
 // String types.
 using strings = std::vector<std::string>;
 
+/// Max depth a cycle detection recursion should go before erroring out
```

Comment:
I don't think solutil is the right place for this, it is a setting for the analyzer. I mean we could also just use literals in both places, not sure if they really need names or need to have the same value.